src/main/java/game/CharacterStats.java

```diff
@@ -0,0 +1,121 @@
+package game;
+
+public class CharacterStats extends Stats {
+
+    public interface Listener {
+
+        default void onDeath() {}
+
+        default void onHealthChange(int oldHealth, int newHealth, int difference) {}
+
+        default void onMaxHealthChange(int oldMaxHealth, int newMaxHealth, int difference) {}
+
+        default void onBaseDamageChange(int oldBaseDamage, int newBaseDamage, int difference) {}
+
+        default void onBaseLuckChange(int oldBaseLuck, int newBaseLuck, int difference) {}
+    }
+
+    private int minHealthPoints;
+    private int maxHealthPoints;
+
+    public CharacterStats(int maxHealthPoints, int baseDamage, int baseLuck) {
+        super(maxHealthPoints, baseDamage, baseLuck);
+        this.minHealthPoints = 0;
+        this.maxHealthPoints = maxHealthPoints;
+    }
+
+    public void applyChange(Stats playerStats) {
+        setHealthPoints(playerStats.healthPoints);
+        setBaseLuck(playerStats.baseLuck);
+        setBaseDamage(playerStats.baseDamage);
+    }
+
+
+    @Override
+    public int getHealthPoints() {
+        return super.getHealthPoints();
+    }
+
+    @Override
+    public void setHealthPoints(int healthPoints) {
+        setHealthPoints(healthPoints, null);
+    }
+
+    public void setHealthPoints(int healthPoints, Listener listener) {
+        int oldHP = this.healthPoints;
+        this.healthPoints += healthPoints;
+
+        if (this.healthPoints > maxHealthPoints) {
+            this.healthPoints = maxHealthPoints;
+        } else if (this.healthPoints < minHealthPoints) {
```

Comment:
here could have just used "else" cause the only other case scenarios to "this.healthPoints > maxHealthPoints" are "<"